File: src/go/radio/api_fly.go
package radio

import "time"

type FlyApi struct {
	*ApiModule
}

func NewFlyApi(rm *RadioManager) *FlyApi {
	return &FlyApi{NewApiModule(rm)}
}

func (api *FlyApi) ListFlyObject() interface{} {
	return api.Send("fly.listFlyObject", nil, "", 3*time.Second, nil)
}

func (api *FlyApi) GetFlyObjectInfo(keyName string) interface{} {
	return api.Send("fly.getFlyObjectInfo", map[string]interface{}{"keyName": keyName}, "", 3*time.Second, nil)
}

func (api *FlyApi) GetFlyObjectCameraImageDown(keyName string) interface{} {
	return api.Send("fly.getFlyObjectCameraImageDown", map[string]interface{}{"keyName": keyName}, "", 3*time.Second, nil)
}

func (api *FlyApi) GetFlyObjectCameraImageFront(keyName string) interface{} {
	return api.Send("fly.getFlyObjectCameraImageFront", map[string]interface{}{"keyName": keyName}, "", 3*time.Second, nil)
}

File: src/lua/radio/api_fly.lua
-- FlyApi implementation in Lua

local api_module = require("radio/api_module")

local FlyApi = {}
FlyApi.__index = FlyApi
setmetatable(FlyApi, { __index = api_module.ApiModule })

function FlyApi.new(rm)
    local self = api_module.ApiModule.new(rm)
    setmetatable(self, FlyApi)
    return self
end

function FlyApi:listFlyObject()
    return self:send("fly.listFlyObject")
end

function FlyApi:getFlyObjectInfo(keyName)
    return self:send("fly.getFlyObjectInfo", { keyName = keyName })
end

function FlyApi:getFlyObjectCameraImageDown(keyName)
    return self:send("fly.getFlyObjectCameraImageDown", { keyName = keyName })
end

function FlyApi:getFlyObjectCameraImageFront(keyName)
    return self:send("fly.getFlyObjectCameraImageFront", { keyName = keyName })
end

return {
    FlyApi = FlyApi
}
